### PR TITLE
Change Balance to CoinBalance and make its references consistent to TokenBalances

### PR DESCRIPTION
**Pull request by igorffs**
## Motivation

After we've introduced TokenBalances (in #582 and #583) the Balances term got ambiguous, so we're changing it to CoinBalance so it can represent this entity more appropriately. 

This also nests CoinBalances into "addresses" the same way TokenBalances did so the table and modules turn into `address_coin_balances` and `Address.CoinBalance` respectively.

### Incompatible Changes
 
- It changes the balances table to address_coin_balances, so it needs a reset in the database.

## Next steps

This PR makes the renaming in modules and database level, but we also have references to "balances" in functions, module attributes and variables that keeps ambiguous that would be nice to change. I'm planning on opening a different PR for that because it might affect our features such as notifications and indexing if there're errors and it enables us to index the blockchain in different environments without worrying with new database changes.

### CODE REVIEW

File: apps/explorer/lib/explorer/chain/address/coin_balance.ex [hunk 1]
```diff
@@ -1,4 +1,4 @@
-defmodule Explorer.Chain.Balance do
```
Comment (igorffs) [inline apps/explorer/lib/explorer/chain/address/coin_balance.ex]: @KronicDeth @feliperenan we have a couple of denormalized columns (`fetched_balance` and 
`fetched_balance_block_number`) that represents a CoinBalance in the `Address` column/schema module. Would it make sense to change it there too?

File: apps/indexer/lib/indexer/address/coin_balances.ex [hunk 2]
```diff
@@ -1,6 +1,6 @@
-defmodule Indexer.Balances do
+defmodule Indexer.Address.CoinBalances do
```
Comment (igorffs) [inline apps/indexer/lib/indexer/address/coin_balances.ex]: @KronicDeth @alexgaribay @feliperenan I think we should think of some different name for this kind of module in the indexer, we have multiple modules that have very similar responsibility and they all have different names, for example:

- `Indexer.Address.CoinBalances`
- `Indexer.Address.TokenBalances`
- `Indexer.AddressExtraction`
- `Indexer.TokenTransfers`

Disclaimer: I'm just pointing that out ☝️, I don't think we need to change this in a hush. Giving names is so difficult D:

File: apps/explorer/lib/explorer/chain/import.ex [hunk 3]
```diff
@@ -10,7 +10,7 @@ defmodule Explorer.Chain.Import do
   alias Explorer.Chain.{
     Address,
     Address.TokenBalance,
-    Balance,
+    Address.CoinBalance,
```
Comment (feliperenan) [inline apps/explorer/lib/explorer/chain/import.ex]: Move it to be before `Address.TokenBalance` to be sorted 🙈 .

File: apps/explorer/lib/explorer/chain/address/coin_balance.ex (hunk 1 above)
Comment (KronicDeth) [reply apps/explorer/lib/explorer/chain/address/coin_balance.ex]: Yes, that would make the naming clear

**Pull request merged**